src/clib/lib/enkf/ecl_config.cpp

```diff
@@ -1,483 +0,0 @@
-   grid, however this does not (in general) apply to the system as a
-   whole. Other objects which internalize pointers (i.e. field_config
-   objects) to an ecl_grid_type instance will be left with dangling
-   pointers; and things will probably die an ugly death. So - changing
-   grid runtime should be done with extreme care.
-*/
-void ecl_config_set_grid(ecl_config_type *ecl_config, const char *grid_file) {
-    if (ecl_config->grid != NULL)
-        ecl_grid_free(ecl_config->grid);
-    ecl_config->grid = ecl_grid_alloc(grid_file);
-}
-
-ui_return_type *ecl_config_validate_grid(const ecl_config_type *ecl_config,
-                                         const char *grid_file) {
-    ui_return_type *ui_return;
-    if (fs::exists(grid_file)) {
-        ecl_file_enum file_type = ecl_util_get_file_type(grid_file, NULL, NULL);
-        if ((file_type == ECL_EGRID_FILE) || (file_type == ECL_GRID_FILE))
-            ui_return = ui_return_alloc(UI_RETURN_OK);
-        else {
-            ui_return = ui_return_alloc(UI_RETURN_FAIL);
-            ui_return_add_error(ui_return,
-                                "Input argument is not a GRID/EGRID file");
-        }
-    } else {
-        ui_return = ui_return_alloc(UI_RETURN_FAIL);
-        ui_return_add_error(ui_return, "Input argument does not exist.");
-    }
-    return ui_return;
-}
-
-const ecl_sum_type *ecl_config_get_refcase(const ecl_config_type *ecl_config) {
-    return ecl_config->refcase;
-}
-
-bool ecl_config_has_refcase(const ecl_config_type *ecl_config) {
-    const ecl_sum_type *refcase = ecl_config_get_refcase(ecl_config);
-    if (refcase)
-        return true;
-    else
-        return false;
-}
-
-bool ecl_config_get_formatted(const ecl_config_type *ecl_config) {
-    return ecl_io_config_get_formatted(ecl_config->io_config);
-}
-bool ecl_config_get_unified_restart(const ecl_config_type *ecl_config) {
-    return ecl_io_config_get_unified_restart(ecl_config->io_config);
-}
-
-bool ecl_config_have_eclbase(const ecl_config_type *ecl_config) {
-    return ecl_config->have_eclbase;
-}
-
-/** Units as specified in the ECLIPSE technical manual */
-const char *ecl_config_get_depth_unit(const ecl_config_type *ecl_config) {
-    switch (ecl_config->unit_system) {
-    case ECL_METRIC_UNITS:
-        return "M";
-    case ECL_FIELD_UNITS:
-        return "FT";
-    case ECL_LAB_UNITS:
-        return "CM";
-    default:
-        util_abort("%s: unit system enum value:%d not recognized \n", __func__,
-                   ecl_config->unit_system);
-        return NULL;
-    }
-}
-
-const char *ecl_config_get_pressure_unit(const ecl_config_type *ecl_config) {
-    switch (ecl_config->unit_system) {
-    case ECL_METRIC_UNITS:
-        return "BARSA";
-    case ECL_FIELD_UNITS:
-        return "PSIA";
-    case ECL_LAB_UNITS:
-        return "ATMA";
-    default:
-        util_abort("%s: unit system enum value:%d not recognized \n", __func__,
```

Comment:
Abort is bad, but we should probably have similar error messages raised. It might be that this error message is already good, but we should add a test for such a message (ie. pytest.raises(ValueError, match="{good message}"):)